Including ratio subplot within differ when comparing

tomeichlersmith opened the pull request:
The denominator for the ratio is whatever histogram is first in the list. This could lead to problems in two situations:
1. The first file in found in the directory (or given on the command line) does not have the histogram requested and so the denominator just becomes the next histogram that is found.
2. There are zero values in the denominator. This will lead to annoying divide-by-zero warnings which we can silence if there are a lot of them.

We try to inherit the color from the histogram in the ratio plot so that the legend applies. This may fail if I misunderstood the return type of the matplotlib `hist` function.

If we want to include error bars, we should probably switch to using [`hist` ](https://hist.readthedocs.io/en/latest/) directly since they have better error bar handling and have ironed out a lot of the common bugs. `uproot` can cast ROOT histograms into `hist.Hist` objects so the loading of already filled histograms is as seamless as it is now.

I am updating _ldmx-sw_, here are the details.

### What are the issues that this addresses?
<!--
_Hint_: Use the phrase '_This resolves #< issue number >_' so that they are linked automatically.
-->
This resolves #1382 .

## Check List
- [x] I successfully compiled _ldmx-sw_ with my developments
- [x] I ran my developments and the following shows that they are successful.
<!-- put plots or some other proof that your developments work and do the intended function -->

tomeichlersmith committed:
draft including ratio subplot within differ when comparing

The denominator for the ratio is whatever histogram is first in the list. 

We try to inherit the color from the histogram in the ratio plot so that the legend applies.

If we want to include error bars, we should probably switch to using `hist` directly since they have better error bar handling and have ironed out a lot of the common bugs.

tvami commented:
It looks promising, here is an example:

<img width="980" alt="Screenshot 2024-08-27 at 14 03 26" src="https://github.com/user-attachments/assets/829906a8-bbd6-4386-abf1-839081d1c653">

I do think tho that we should not connect the points in the ratio inset!

tvami commented:
<img width="860" alt="Screenshot 2024-08-27 at 14 06 20" src="https://github.com/user-attachments/assets/ee109974-3a0b-4b4e-9d59-a3236bc56937">

In this example, we have zero bins for the first histo, and the following histos too --> I think the current behavior is great as it is for that

tvami reviewed (changes requested):
The only thing I'd like to change is the connecting the dots in the ratio inset

tomeichlersmith committed:
change connected dots to a scatter plot

We could get a scatter plot by altering the `format` passed to `plot` but `scatter` has nice defaults.

tvami commented:
Looks very nice when the binning size is reasonable
<img width="1029" alt="Screenshot 2024-08-28 at 13 08 01" src="https://github.com/user-attachments/assets/3bc122be-f716-49b1-baef-f047ba98facf">

tvami commented:
when the binning is tiny, it's a bit messy:
<img width="1195" alt="Screenshot 2024-08-28 at 13 08 58" src="https://github.com/user-attachments/assets/4043ad9b-3a20-45ff-a4c9-6b20e84da265">

tvami reviewed (approved):
I think this looks good now, my comment for the binning is beyond this PR

tvami commented:
Plots were made by Gregorio @ohghvstt
